libp2p/crypto/crypto.nim

```diff
@@ -130,17 +136,23 @@ proc random*(t: typedesc[KeyPair], scheme: PKScheme,
     result.seckey.skkey = pair.seckey
     result.pubkey.skkey = pair.pubkey
 
-proc getKey*(key: PrivateKey): PublicKey =
+proc getKey*(key: PrivateKey): CryptoResult[PublicKey] =
   ## Get public key from corresponding private key ``key``.
-  result = PublicKey(scheme: key.scheme)
+  var res = PublicKey(scheme: key.scheme)
```

Comment:
I'd move this into the `if`:
```
ok(PublicKey(scheme: key.scheme, rsakey: key.rsakey.getKey()))
```

Reply:
Enum type compilation fails if I do that cos cannot infer even if scheme is there :/

```
ok(PrivateKey(scheme: scheme, rsakey: rsakey))
```
gives:
```
Error: cannot prove that it's safe to initialize 'rsakey' with the runtime value for the discriminator 'scheme'
```

What works is 
```
ok(PrivateKey(scheme: RSA, rsakey: rsakey))
```

Reply:
Inside a `case/of` works btw, the `if` was the issue